File: Advent of Code 2022/src/DayOne.java
import java.io.BufferedReader;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;

public class DayOne {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new FileReader("data/dayone.txt"));

        String line = br.readLine();

        int temp = 0;
        int first = 0;
        int second = 0;
        int third = 0;

        while(line != null) {
            if (line.equals("")) {
                if (temp > first) {
                    third = second;
                    second = first;
                    first = temp;
                } else if (temp > second) {
                    third = second;
                    second = temp;
                } else if (temp > third) {
                    third = temp;
                }

                temp = 0;
            } else {
                temp += Integer.parseInt(line);
            }

            line = br.readLine();
        }

        if (temp != 0) {
            if (temp > first) {
                third = second;
                second = first;
                first = temp;
            } else if (temp > second) {
                third = second;
                second = temp;
            } else if (temp > third) {
                third = temp;
            }
        }

        System.out.println(first + second + third);
    }
}

File: Advent of Code 2022/js/DayOne.js
function PartOne() {
    var fs = require("fs");

    const data = fs.readFileSync("../data/dayone.txt", "utf8").split("\n");

    let temp = 0;
    let max = 0;

    for (let i = 0; i < data.length; i++) {
        if (data[i] === "") {
            if (temp > max) {
                max = temp;
            }

            temp = 0;
        } else {
            temp += parseInt(data[i]);
        }
    }

    return max;
}

function PartTwo() {
    var fs = require("fs");

    const data = fs.readFileSync("../data/dayone.txt", "utf8").split("\n");

    let temp = 0;
    let first = 0;
    let second = 0;
    let third = 0;

    for (let i = 0; i < data.length; i++) {
        if (data[i] === "") {
            if (temp > first) {
                third = second;
                second = first;
                first = temp;
            } else if (temp > second) {
                third = second;
                second = temp;
            } else if (temp > third) {
                third = temp;
            }

            temp = 0;
        } else {
            temp += parseInt(data[i]);
        }
    }

    if (temp != 0) {
        if (temp > first) {
            third = second;
            second = first;
            first = temp;
        } else if (temp > second) {
            third = second;
            second = temp;
        } else if (temp > third) {
            third = temp;
        }
    }

    return first + second + third;
}

const partOne = PartOne();
const partTwo = PartTwo();

console.log("Part One: " + partOne, "\nPart Two: " + partTwo);
